generated/nidaqmx/nidaqmx_service.cpp

```diff
@@ -12932,7 +12932,7 @@ namespace nidaqmx_grpc {
       int32 samps_per_chan_read {};
       auto status = library_->ReadAnalogF64(task, num_samps_per_chan, timeout, fill_mode, read_array, array_size_in_samps, &samps_per_chan_read, reserved);
       if (!status_ok(status)) {
-        return ConvertApiErrorStatusForTaskHandle(context, status, task);
+        return ConvertApiErrorStatusWithReadParametersForTaskHandle(context, status, samps_per_chan_read, task);
       }
```

Comment:
On further thought: how much value do the helper functions add, compared to this?
```
      if (!status_ok(status)) {
        context->AddTrailingMetadata("ni-samps-per-chan-read", std::to_string(samps_per_chan_read));
        return ConvertApiErrorStatusForTaskHandle(context, status, task);
      }
```

Duplicating this line adds negligible bloat ((1 virtual method call + 1 function call) x 20-30 functions) and it's all in the `if (!status_ok(status))` path.

Reply:
As per your suggestion, a simple `boolean` will not be enough as we would still need the logic to figure out the read/write non-scalar functions. To do so, we need the additional metadata in `functions.addon`. Even then the parameter to be used for populating the error varies in each function. To simplify this, we have added a property called `return_on_error_key` in the functions metadata under the corresponding parameter. This will ensure simplicity and the generic approach that we are looking for.

Reply:
I agree, specifying the gRPC metadata key name in the API metadata is more generic.